File: Engine/src/main/java/org/etieskrill/engine/graphics/gl/shader/impl/AnimationShader.java
package org.etieskrill.engine.graphics.gl.shader.impl;

import io.github.etieskrill.injection.extension.shaderreflection.ReflectShader;
import org.etieskrill.engine.graphics.gl.shader.ShaderProgram;

import java.util.List;

@ReflectShader
public class AnimationShader extends ShaderProgram {
    public AnimationShader() {
        super(List.of("Animation.vert", "Animation.frag"));
    }

    @Override
    protected void setUniformDefaults() {
        AnimationShaderKt.setShowBoneSelector(this, 4);
        AnimationShaderKt.setShowBoneWeights(this, false);
    }
}

File: Engine/src/commonMain/kotlin/org/etieskrill/engine/graphics/shader/impl/AnimationShader.kt
package org.etieskrill.engine.graphics.shader.impl

import io.github.etieskrill.injection.extension.shader.reflection.ReflectShader
import org.etieskrill.engine.graphics.shader.Shader

@ReflectShader
class AnimationShader : Shader(listOf("Animation.vert", "Animation.frag")) {
//    @Override
//    protected void setUniformDefaults() {
//        AnimationShaderKt.setShowBoneSelector(this, 4);
//        AnimationShaderKt.setShowBoneWeights(this, false);
//    }
}
